src/search/landmarks/landmark_cost_partitioning_heuristic.cc

```diff
@@ -0,0 +1,159 @@
+        cerr << "conditional effects not supported by the landmark "
+             << "generation method" << endl;
+        utils::exit_with(ExitCode::SEARCH_UNSUPPORTED);
+    }
+
+    if (opts.get<bool>("optimal")) {
+        lm_cost_assignment =
+            utils::make_unique_ptr<LandmarkEfficientOptimalSharedCostAssignment>(
+                task_properties::get_operator_costs(task_proxy),
+                *lm_graph, opts.get<lp::LPSolverType>("lpsolver"));
+    } else {
+        lm_cost_assignment =
+            utils::make_unique_ptr<LandmarkUniformSharedCostAssignment>(
+                task_properties::get_operator_costs(task_proxy),
+                *lm_graph, opts.get<bool>("alm"));
+    }
+}
+
+int LandmarkCostPartitioningHeuristic::get_heuristic_value(
+    const State &/*state*/) {
+    double epsilon = 0.01;
+
+    double h_val = lm_cost_assignment->cost_sharing_h_value(*lm_status_manager);
+    if (h_val == numeric_limits<double>::max()) {
+        return DEAD_END;
+    } else {
+        return static_cast<int>(ceil(h_val - epsilon));
+    }
+}
+
+bool LandmarkCostPartitioningHeuristic::dead_ends_are_reliable() const {
+    return true;
+}
+
+class LandmarkCostPartitioningHeuristicFeature : public plugins::TypedFeature<Evaluator, LandmarkCostPartitioningHeuristic> {
+public:
+    LandmarkCostPartitioningHeuristicFeature() : TypedFeature("lmcp") {
+        document_title("Landmark cost partitioning heuristic");
+        document_synopsis(
+            "See the papers" +
+            utils::format_conference_reference(
+                {"Erez Karpas", "Carmel Domshlak"},
+                "Cost-Optimal Planning with Landmarks",
+                "https://www.ijcai.org/Proceedings/09/Papers/288.pdf",
+                "Proceedings of the 21st International Joint Conference on "
+                "Artificial Intelligence (IJCAI 2009)",
+                "1728-1733",
+                "AAAI Press",
+                "2009") +
+            "and" +
+            utils::format_conference_reference(
+                {"Emil Keyder and Silvia Richter and Malte Helmert"},
+                "Sound and Complete Landmarks for And/Or Graphs",
+                "https://ai.dmi.unibas.ch/papers/keyder-et-al-ecai2010.pdf",
+                "Proceedings of the 19th European Conference on Artificial "
+                "Intelligence (ECAI 2010)",
+                "335-340",
+                "IOS Press",
+                "2010"));
+
+        LandmarkHeuristic::add_options_to_feature(*this);
+        add_option<bool>(
+            "optimal",
+            "use optimal (LP-based) cost sharing",
+            "false");
+        add_option<bool>("alm", "use action landmarks", "true");
+        lp::add_lp_solver_option_to_feature(*this);
+
+        document_note(
+            "Optimal search",
+            "You probably also want to add this heuristic as a lazy_evaluator "
+            "in the A* algorithm to improve heuristic estimates.");
```

Comment:
I don't like the phrasing here. Suggestion:
"Usage with A*"
"We recommend to add this heuristic as lazy_evaluator when using it in the A* algorithm. This way, the heuristic is recomputed before a state is expanded, leading to improved estimates that incorporate all knowledge gained from paths that were found after the state was inserted into the open list.

Reply:
Did you check that this is what happens when setting lazy evaluators? I'm always confused because I think there are two different things described in the literature describing lazy evaluation. One fits to what you suggest to write here. The other thing is that states are inserted into the open list with the heuristic estimates of their parents rather than their own ones to avoid computing the heuristic for every generated state, but only compute the heuristic value once a state is expanded.

Reply:
I agree the naming is unfortunate to say the least :). The second thing you mention is *lazy_search*. Lazy_evaluator is only implemented in eager_search, and the only thing it does different is that when a node is removed from the open list it will recompute the heuristic if necessary (checked with the h_dirty flag) and push the node back in open if the heuristic value changed.

Reply:
OK, that sounds very familiar, then I'll adapt the comment as you suggested.